### PR TITLE
op lib wrappers

### PR DESCRIPTION
**Pull request by 1sanz**
all , any , argmax

### CODE REVIEW

File: plaidml2/op/op_test.cc [hunk 1]
```diff
@@ -155,10 +229,9 @@ TEST(Op, Sum) {
 
 TEST(Op, Squeeze) {
   auto A = Placeholder(PLAIDML_DATA_FLOAT32, {32, 1, 4, 1}, "A");
-  auto t = op::squeeze( //
-      A,             // tensor to squeeze
-      {1, 3}         // axes to squeeze
-  );
+  auto t = op::squeeze(  //
+      A,                 // tensor to squeeze
+      {1, 3} /* axes to squeeze*/);
```
Comment (dgkutnic) [inline plaidml2/op/op_test.cc]: The formatting of these was fixed in master, could you resolve these in your branch?

File: plaidml2/op/op_test.cc [hunk 2]
```diff
@@ -193,10 +266,9 @@ TEST(Op, Reshape) {
 
 TEST(Op, Tile) {
   auto A = Placeholder(PLAIDML_DATA_FLOAT32, {10, 20}, "A");
-  auto t = op::tile( //
-      A,             // tensor to tile
-      {5, 4}         // tiling factors
-  );
+  auto t = op::tile(  //
+      A,              // tensor to tile
+      {5, 4} /* tiling factors*/);
```
Comment (dgkutnic) [inline plaidml2/op/op_test.cc]: Same as my previous comment -- these should be correct in master.

File: plaidml2/op/op_test.cc (hunk 1 above)
Comment (1sanz) [reply plaidml2/op/op_test.cc]: Done :)

File: plaidml2/op/op.h [hunk 3]
```diff
@@ -37,6 +37,18 @@ inline edsl::Tensor abs(const edsl::Tensor& I_or_O  //
   auto args = edsl::make_tuple(I_or_O);
   return details::op("abs", args).as_tensor();
 }
+inline edsl::Tensor all(const edsl::Tensor& I, const edsl::Value& axes = edsl::None(), bool keepdims = false) {
+  auto args = edsl::make_tuple(I, axes, keepdims);
+  return details::op("all", args).as_tensor();
+}
+inline edsl::Tensor any(const edsl::Tensor& I, const edsl::Value& axes = edsl::None(), bool keepdims = false) {
```
Comment (flaub) [inline plaidml2/op/op.h]: Could you add a newline between multiple functions?

File: plaidml2/op/op_test.cc [hunk 4]
```diff
@@ -53,6 +53,80 @@ TEST(Op, abs) {
 )"));
 }
 
+TEST(Op, all) {
```
Comment (flaub) [inline plaidml2/op/op_test.cc]: To be consistent, the test name is `CamelCase`

File: plaidml2/op/op.h [hunk 5]
```diff
@@ -30,14 +30,33 @@ inline edsl::Value op(const std::string& name, const edsl::Value& args) {
...
+  auto args = edsl::make_tuple(I, axes, keepdims);
+  return details::op("any", args).as_tensor();
+}
+
+inline edsl::Tensor argmax(const edsl::Tensor& I, const edsl::Value& axes = edsl::None()) {
+  auto args = edsl::make_tuple(I, axes);
+  return details::op("argmax", args).as_tensor();
+}
+
+inline edsl::Tensor max(const edsl::Tensor& I,  // NOLINT(build/include_what_you_use)
```
Comment (flaub) [inline plaidml2/op/op.h]: Why does this line need NOLINT?

File: plaidml2/op/op.h [hunk 6]
```diff
@@ -30,14 +30,33 @@ inline edsl::Value op(const std::string& name, const edsl::Value& args) {
 
 }  // namespace details
 
-//  return absolute value of tensor
-
 inline edsl::Tensor abs(const edsl::Tensor& I_or_O  //
```
Comment (flaub) [inline plaidml2/op/op.h]: We should probably rename this to `I` (not `I_or_O`)

File: plaidml2/op/op_test.cc [hunk 7]
```diff
@@ -53,6 +53,96 @@ TEST(Op, abs) {
...
+  _X5[] = >(I[x0, x1, x2, x3] == _X0[] ? _X4[x0, x1, x2, x3]);
+  _X6 = 32;
+  _X7 = as_uint(_X5, _X6);
+}
+)"));
+}
+
+TEST(Op, Max) {
+  auto I = Placeholder(PLAIDML_DATA_FLOAT32, {1, 224, 224, 3}, "I");
+  auto max = op::max(I);  // NOLINT(build/include_what_you_use)
```
Comment (flaub) [inline plaidml2/op/op_test.cc]: Why do we need NOLINT?

**Pull request merged**